### DISCUSSION TITLE
Validation doesn’t work on unmounted part of form

### DISCUSSION THREAD

**Discussion by mayase** [react-hook-form/react-hook-form#9108] [Q&A] [answered]
It seems like unmounted parts of form are not being validated by trigger/submit
See repro below

### Version Number

7.36.1

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-usefieldarray-forked-n43hhm?file=/src/index.js

### Steps to reproduce

1. Launch provided codesandbox
2. There is a form with two sets of field. First set is active, second is hidden, you can switch by clicking on respective button. On the top part of the screen there is "form is valid" text. Validation happens on each change (with `watch` and `trigger`)
4. Remove text "Luo from fieldSet1" from the first field
5. Form is invalid, because this field is required. You can see the text changed to "form is invalid"
6. click on button "FIELDSET2"
7. Another set of fields appears, text changes to "form is valid" <-- **that is strange**
8. On submit data is printed, you can see in the console, that we have two sets of fields

### Expected behaviour

The from should be in invalid state on step 7

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
unmounted input will not get validated if ths is an issue you can use scheme validation library instead as it validate at the form level.

**Reply by mayase** [reply]
Thanks, you are referring to `resolver` option? e.g. for yup it will be like this
```
useForm({
    resolver: yupResolver(schema),
  });
```

And then data itself will be validated, not just mounted inputs?

doc ref to be precise https://github.com/react-hook-form/resolvers#quickstart

**Reply by bluebill1049** [reply]
correct
